dialogflow/stopwatch/Stopwatch/AppDelegate.swift

```diff
@@ -14,16 +14,32 @@
 // limitations under the License.
 //
 import UIKit
+import FirebaseFunctions
+import Firebase
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
-  
   var window: UIWindow?
+  var hasRequestedForToken = false
+  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions:[UIApplicationLaunchOptionsKey :Any]? = nil) -> Bool {
+    FirebaseApp.configure()
+    generateAccessToken()
+    return true
+  }
   
-  func application
-    (_ application: UIApplication,
-     didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey : Any]? = nil)
-    -> Bool {
-      return true
+  func generateAccessToken() {
+    //Block the UI/ show a pop up message / a diff screen
```

Comment:
Might be worth mentioning that this uses Firebase Auth and you can insert any auth protocol that they offer.